### PR TITLE
PredictionFunction becomes PredictionEngine

### LINKED ISSUE #1761: Name of MakePredictionFunction is confusing (to me).
I was following the start example in the Readme, and when I came to the line with MakePredictionFunction, I got a bit confused and had to go over it a few times before I could continue:

> Now from the model we can make inferences (predictions):
> 
> ```C#
> var predictionFunc = model.MakePredictionFunction<SentimentData, SentimentPrediction>(mlContext);
> var prediction = predictionFunc.Predict(new SentimentData
> {
>     SentimentText = "Today is a great day!"
> });
> Console.WriteLine("prediction: " + prediction.Prediction);
> ```

I expected the return value of `MakePredictionFunction()` to a function-type thing, but instead it was an object.  This was hard for me not because of pedantic correctness, but rather from a usability standpoint as a beginner trying to keep my concepts straight (estimators, transformers, models, etc.) this took me a step back and I had to go over the section several times to make sure I hadn't missed something.

Is there a better name that would be less confusing?
- `MakePredictionObject()`
- `MakePredictionEngine()`
- `MakePredictor()`
- ?

### LINKED ISSUE #1819: CreatePredictionEngine<TSrc, TDst> is internal now. What is replacement?
The CreatePredictionEngine method is internal now. What is replacement for trainedModel (ITransfomer) and scheme (SchemaDefinition)? MakePredictionFunction can't use SchemaDefinition and makes exception for me.

### PR DESCRIPTION
**Pull request by Zruty0**
Fixes #1761 
Fixes #1819 

Renamed MakePredictionFunction to CreatePredictionEngine, and exposed the schema comprehension parameters to the factory method.

Added a method to `mlContext.Model` to create a prediction engine.

### CODE REVIEW

File: src/Microsoft.ML.Data/Model/PredictionEngineExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML
+{
+    /// <summary>
+    /// Extension methods to create a prediction engine.
+    /// </summary>
+    public static class PredictionEngineExtensions
+    {
+        public static PredictionEngine<TSrc, TDst> CreatePredictionEngine<TSrc, TDst>(this ITransformer transformer,
```
Comment (sfilipi) [inline src/Microsoft.ML.Data/Model/PredictionEngineExtensions.cs]: >public [](start = 8, length = 6)

XML doc.

File: src/Microsoft.ML.Transforms/HashJoiningTransform.cs [hunk 2]
```diff
@@ -542,40 +541,23 @@ private ValueGetter<VBuffer<uint>> ComposeGetterVecToVec<TSrc>(Row input, int ii
...
             var hashSeed = _exes[iinfo].HashSeed;
             bool ordered = _exes[iinfo].Ordered;
-            TSrc[] denseValues = null;
+            var denseSource = default(VBuffer<TSrc>);
             return
                 (ref VBuffer<uint> dst) =>
                 {
                     getSrc(ref src);
                     Host.Check(src.Length == expectedSrcLength);
-                    ReadOnlySpan<TSrc> values;
```
Comment (sfilipi) [inline src/Microsoft.ML.Transforms/HashJoiningTransform.cs]: is this Ivan's other commit?

File: src/Microsoft.ML.Transforms/HashJoiningTransform.cs [hunk 3]
```diff
@@ -644,11 +611,11 @@ private ValueGetter<uint> ComposeGetterVecToOne<TSrc>(Row input, int iinfo)
         /// </summary>
         private HashDelegate<TSrc> ComposeHashDelegate<TSrc>()
         {
-            // REVIEW: Add a specialized hashing for ints, once numeric bin mapper is done http://sqlbuvsts01:8080/Main/Advanced%20Analytics/_workitems/edit/5823788
```
Comment (sfilipi) [inline src/Microsoft.ML.Transforms/HashJoiningTransform.cs]: does the description on this workitem needs to be logged in github?

**Comment by TomFinley** [comment]
I've tried restarting your failing build a few times. You are being plagued by what appears to be a spurious test failure on PFI clustering test, on Mac OS debug. I have attempted to try to fix it according to my best guesses in #1926. If we can commit that PR, great, but if my guess is right and you find it more expeditious to merge my small change into your PR just to unblock yourself that might be a better policy. Thanks @Zruty0 !

**Pull request merged**